### ISSUE TITLE
Dashboard frontend

### ISSUE THREAD

**Issue by JortvD** [JortvD/meta#1]
How would we show the dashboard to users? With scalability in mind this can be done in multiple ways:

- A simple webserver created by the dashboard that hosts the frontend.
- A seperate frontend service that connects to the dashboard via an API. This service is then also by default included with the dashboard backend. (Has my preference)
- Etc.

**Comment by theotterlord** [comment]
> - A seperate frontend service that connects to the dashboard via an API. This service is then also by default included with the dashboard backend. (Has my preference)

I agree here.

- An API is less resource heavy on the backend service, as the display is handled on the client side. A web server would require all html/css/js to be served each time a client uses the service. 
- It also allows hosts/developers to create custom front end services without modifying the backend. 

However, 

- If we choose to support access from mobile/tablet platforms, we'll need to provide an app for each major platform. A web app (PWA) hosted by the server would be so much easier to build and maintain.


One way to do this could be to implement an API (with docs for developers) and offer a frontend app by default. Then, create a plugin progressive web app (adapted for all major platforms) that users can optionally install. There are a few benifits to this. 

- Less resource heavy by default (without plugin)
- Allow hosts/devs to create custom solutions
- Provides a service for all devices (via PWA) plus a dedicated desktop app
- The PWA can be created using the same codebase as the desktop app (using Electron)

I don't know much about web host managers and if mobile support is even a thing that people want, but from what I do know, this seems like the best way forward. Let me know what you think 😄

**Comment by JortvD** [comment]
Yes, I agree having a sole codebase for both a mobile and desktop app (and maybe even the website?) is a lot easier (and less work).

> One way to do this could be to implement an API (with docs for developers)...

And I am all for designing the code in such way that there are no internal API's for our frontend alone.

> ...and offer a frontend app by default.

Do you mean the frontend site?

> I don't know much about web host managers and if mobile support is even a thing that people want,...

I think mobile support will really help on the fly observervation and modification. It will take some extra thinking for the design philosophy though.

**Comment by theotterlord** [comment]
> Do you mean the frontend site?

Kind of. I mean an Electron packaged desktop application using the same code as the web app would. The desktop app would automatically work as the API is builtin. However, the web app would have to be installed as a plugin (or set up on another server and authenticated somehow) as some people may just want the desktop app (or a custom installation). 

> I think mobile support will really help on the fly observervation and modification. It will take some extra thinking for the design philosophy though.

Maybe, but we just need to put ourselves in our users position. Why would we use the mobile app instead of using the desktop app. Maybe there is some constantly changing data we wish to keep an eye on, update some server software quickly, etc. For now, we don't need to worry about it too much. Once we know what data will be available, then we can decide what data to show and how to show it.

**Comment by JortvD** [comment]
I agree. It may also be smarter to not start overexerting ourselves by starting with the mobile/destop app. By starting with a simple webapp we can work on those later. Using an API though makes our future options a lot broader than when we don't. And we should of course also keep responsiveness in mind while making design decisions.
